src/compiler/checker.ts

```diff
@@ -11336,6 +11336,34 @@ namespace ts {
                 return type;
             }
 
+            function isTypePresencePossible(type: Type, propName: string, shouldHaveProperty: boolean) {
+                const prop = getPropertyOfType(type, propName);
+                if (!prop) {

```

Comment:
Two edits in this body - rename `shouldHaveProperty` to `assumeTrue`  and reduce the `if` to 
```
if (prop) {
  return (prop.flags & SymbolFlags.Optional) ? true : assumeTrue;
}
return !assumeTrue;
```